netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -323,10 +323,13 @@ void returnProcessedBytes(Http2Stream stream, int bytes) {
   }
 
   private void onHeadersRead(int streamId, Http2Headers headers, boolean endStream) {
-    NettyClientStream.TransportState stream = clientStream(requireHttp2Stream(streamId));
-    stream.transportHeadersReceived(headers, endStream);
-    if (keepAliveManager != null) {
-      keepAliveManager.onDataReceived();
+    // Stream 1 is reserved for the Upgrade response, so we should ignore its headers here:
+    if (streamId != 1) {
```

Comment:
Let's reference https://github.com/netty/netty/blob/netty-4.1.25.Final/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java#L47 here instead of just plain `1`.